### ISSUE TITLE
Ack, Server Error on makeabilitylab-test talks page

### ISSUE THREAD

**Issue by jonfroehlich** [jonfroehlich/makeabilitylabwebsite#535]
I seemingly can load other pages though...

![image](https://user-images.githubusercontent.com/1621749/43656344-f922ac5a-9706-11e8-98d2-094a183d239f.png)

**Comment by aileenzeng** [comment]
Here's a snippet from the log -- looks like there's something funky going on w/ `cap_title`:

```
DEBUG 2018-08-03 13:21:49,089 base 129 140006861907712 Exception while resolving variable 'get_title' in template 'website/talks.html'.
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 864, in _resolve_lookup
    current = current()
  File "/code/website/models.py", line 675, in get_title
    cap_title = ' '.join(s[0].upper() + s[1:] for s in self.title.split(' '))
  File "/code/website/models.py", line 675, in <genexpr>
    cap_title = ' '.join(s[0].upper() + s[1:] for s in self.title.split(' '))
IndexError: string index out of range
ERROR 2018-08-03 13:21:49,090 exception 129 140006861907712 Internal Server Error: /talks/
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/django/core/handlers/exception.py", line 35, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py", line 128, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/lib/python3.6/site-packages/django/core/handlers/base.py", line 126, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/code/website/views.py", line 218, in talks
    return render(request, 'website/talks.html', context)
  File "/usr/local/lib/python3.6/site-packages/django/shortcuts.py", line 36, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "/usr/local/lib/python3.6/site-packages/django/template/loader.py", line 62, in render_to_string
    return template.render(context, request)
  File "/usr/local/lib/python3.6/site-packages/django/template/backends/django.py", line 61, in render
    return self.template.render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 175, in render
    return self._render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 167, in _render
    return self.nodelist.render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 943, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 910, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/loader_tags.py", line 155, in render
    return compiled_parent._render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 167, in _render
    return self.nodelist.render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 943, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 910, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/loader_tags.py", line 67, in render
    result = block.nodelist.render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 943, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 910, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/defaulttags.py", line 211, in render
    nodelist.append(node.render_annotated(context))
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 910, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 993, in render
    output = self.filter_expression.resolve(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 676, in resolve
    obj = self.var.resolve(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/loader_tags.py", line 67, in render
    result = block.nodelist.render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 943, in render
    bit = node.render_annotated(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 910, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/defaulttags.py", line 211, in render
    nodelist.append(node.render_annotated(context))
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 910, in render_annotated
    return self.render(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 993, in render
    output = self.filter_expression.resolve(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 676, in resolve
    obj = self.var.resolve(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 802, in resolve
    value = self._resolve_lookup(context)
  File "/usr/local/lib/python3.6/site-packages/django/template/base.py", line 864, in _resolve_lookup
    current = current()
  File "/code/website/models.py", line 675, in get_title
    cap_title = ' '.join(s[0].upper() + s[1:] for s in self.title.split(' '))
  File "/code/website/models.py", line 675, in <genexpr>
    cap_title = ' '.join(s[0].upper() + s[1:] for s in self.title.split(' '))
IndexError: string index out of range
ERROR 2018-08-03 13:21:49,090 basehttp 129 140006861907712 "GET /talks/ HTTP/1.1" 500 27
```

**Comment by aileenzeng** [comment]
@mechanicjay Do you think you could also take a look at this if you have time?

**Comment by mechanicjay** [comment]
I'll be able to take a look this afternoon.

--Jason

On August 3, 2018 10:46:34 AM PDT, Aileen Zeng <notifications@github.com> wrote:
>@mechanicjay Do you think you could also take a look at this if you
>have time?
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/jonfroehlich/makeabilitylabwebsite/issues/535#issuecomment-410327352

-- 
Sent from my Android phone with K-9 Mail. Please excuse my brevity.

**Comment by jonfroehlich** [comment]
If it’s related to cap title code, that was added by the high school interns very early on and perhaps not properly tested. Before we bother Jason, this code should be checked and verified that it works in a local dev environment. Johnson or Shiven should do this.

**Comment by mechanicjay** [comment]
Thanks Jon, will hold off till I hear back from you guys on this.

**Comment by jonfroehlich** [comment]
I believe either @ShivenBhatt or @johnsonkuang wrote the code for the cap_title stuff and should test and fix this. See the error log that @aileenzeng posted here: https://github.com/jonfroehlich/makeabilitylabwebsite/issues/535#issuecomment-410322142. Key portion:

```
File "/code/website/models.py", line 675, in get_title
    cap_title = ' '.join(s[0].upper() + s[1:] for s in self.title.split(' '))
  File "/code/website/models.py", line 675, in <genexpr>
    cap_title = ' '.join(s[0].upper() + s[1:] for s in self.title.split(' '))
IndexError: string index out of range
```

Did you guys test this code locally by uploading and trying to access talk pdf and pptx files? If not, please do so and report back.

@mechanicjay, sorry for the preemptive bother. We may still need you on this but not for now.

**Comment by johnsonkuang** [comment]
Tested locally and all the following cases worked:
- title with special characters ( exclamation mark, quotation mark, colon, semicolon, back and forward slashes"
- title with leading spaces

There was nothing wrong in rendering them. 
To create the talks, I simply editted the titles but gave arbitrary data for the rest of the required fields.

**Comment by jonfroehlich** [comment]
@johnsonkuang, you've got to prioritize this one... did you manage to finish the new import stuff? If so, can you import all of the talks off of makeabilitylab-test and run your server locally with that data--do you see the same Server Error (500) problem? I have a feeling that the cap_title code is barfing on some weird talk title (and the cap_title code is new(er)).

**Comment by jonfroehlich** [comment]
How we coming on this? What if we simply set `cap_title = self.title` for now and then you try to track down the fix on your own. This way, at least the page will load.

**Comment by johnsonkuang** [comment]
The suspicions that I had have been disproved. Initially, I thought single characters could be the cause behind the error; however, the method was able to convert the single characters to capital in a plain python environment with no problem. I did, however, find that any leading spaces or ending spaces would cause problems due to the existence of null items in the list after running split by space. Django was keen on this and trims all the excess spaces before and after the first and last string character when uploading via admin. Special characters were also not a problem.

Do you have any more suggestions for what else to test?

**Comment by johnsonkuang** [comment]
Server error resolved, however a new issue has resulted #569. But we've made forward progress.
